redis/src/cluster_async/mod.rs

```diff
@@ -699,20 +711,30 @@ where
 
     async fn try_cmd_request(
         cmd: Arc<Cmd>,
-        func: fn(C, Arc<Cmd>) -> RedisFuture<'static, Response>,
         redirect: Option<Redirect>,
-        routing: Option<RoutingInfo>,
+        routing: CommandRouting<C>,
         response_policy: Option<ResponsePolicy>,
         core: Core<C>,
         asking: bool,
     ) -> (OperationTarget, RedisResult<Response>) {
+        let routing = match routing {
+            CommandRouting::Route(routing) => routing,
+            CommandRouting::Connection { addr, conn } => {
+                if !asking {
+                    let mut conn = conn.await;
+                    let result = conn.req_packed_command(&cmd).await.map(Response::Single);
+                    return (addr.into(), result);
+                }
+                None
```

Comment:
> so, if 'asking', we get a random node? why?

No, if `asking` then we'll get the redirected connection when calling `get_connection`.

> If we already have the connection, why won't we just do something like 

Because then we'd use the old connection instead of getting the redirected connection.